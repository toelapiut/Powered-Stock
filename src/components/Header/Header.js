import React from 'react';
import PropTypes from 'prop-types';
import DateInput from '../../components/DateInput';
import styles from './header.module.css';



export const Header = ({name, date, isOpen, onChangeDates, ticker, start, end, onOpenCalendar}) => {
  return (
    <div className={styles.container}>
      <div>
        <h1 className={styles.ticker}>{ticker}</h1>
        <p className={styles.name}>{name}</p>
      </div>
      <DateInput
        date={date}
        isOpen={isOpen}
        onChangeDates={onChangeDates}
        start={start}
        end={end}
        onOpenCalendar={onOpenCalendar}
      />
    </div>
  );
};

Header.propTypes = {
  date: PropTypes.shape({
    start: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    end: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ])
  }),
  name:PropTypes.string.isRequired,
  ticker:PropTypes.string.isRequired,
  start: PropTypes.string,
  end: PropTypes.string,
  isOpen: PropTypes.bool.isRequired,
  onChangeDates: PropTypes.func.isRequired,
  onOpenCalendar: PropTypes.func.isRequired,
};

