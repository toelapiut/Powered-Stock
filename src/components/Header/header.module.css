.container {
    @apply bg-koamaru pl-8 pr-4 py-4 rounded-lg mb-5 mt-6 flex flex-row items-center justify-between
}
.details{

}

.ticker{
    @apply text-3xl font-black text-white leading-10
}

.name{
    @apply text-sm font-bold text-liberty
}