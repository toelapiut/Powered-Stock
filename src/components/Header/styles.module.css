.container {
    @apply pt-10 px-10 mb-5
}

.title {
    @apply text-3xl font-black text-white
}


.header {
    @apply text-koamaru font-bold text-base;
}