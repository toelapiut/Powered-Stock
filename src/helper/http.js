import axios from 'axios';

export const quandl = axios.create({
  baseURL: process.env.process.env.REACT_APP_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Allow': 'GET, PUT, HEAD, OPTIONS'
  }
});

quandl.interceptors.request.use((config) => {
  config.params = config.params || {};
  config.params['key'] = process.env.REACT_APP_API_KEY;
  return config;
});
