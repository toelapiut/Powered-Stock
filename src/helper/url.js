export const url = {
  markets: 'datatables'
};

export const types = {
  ZACKS_CP: 'ZACKS/CP',
  ZACKS_MT: 'ZACKS/MT'
};