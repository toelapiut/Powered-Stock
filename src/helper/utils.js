import humps from 'humps';
import checkPropTypes from 'check-prop-types';
import {timeFormat} from 'd3-time-format';
import {bisector} from 'd3-array';
import moment from 'moment';

export const transpose = (rows, cols) => {
  /*
   * check if the row and col are empty and throw an error to
   * inform of the necessary requirements
   */
  if (typeof rows === 'undefined' || typeof cols === 'undefined') {
    throw new Error('row and col can not be undefined');
  }

  /*
   * throw error if the array isn't an array or col isn't an array
   */
  if (!Array.isArray(rows)) throw new Error('row must be  array');
  if (!Array.isArray(cols)) throw new Error('col must be  array');

  let dataset = [];

  for (let row of rows) {
    /*
     * check the data type before iterating through the matrix
     */
    if (!Array.isArray(row)) throw new Error('row must be a matrix row: [[a, b]]');
    let obj = {};
    row.forEach((item, index) => {
      obj[cols[index]['name']] = item;
    });
    dataset.push(obj);
  }

  return dataset;
};


export const transposeStock = (rows) => {
  let data = [];
  for (let row of rows) {
    let obj = {};
    const [date, close] = row;
    obj['date'] = date;
    obj['close'] = close;
    data.push(obj);
  }
  return data;
};

export const toCamelCase = (item) => {
  if (typeof item === 'undefined') {
    return item;
  }
  return humps.camelizeKeys(item);
};

export const checkPositivity = (num) => {
  if (typeof num === 'undefined') throw new Error('argument must not be empty');
  if (typeof num !== 'number') throw new Error('argument is not a number');
  let check = Math.sign(Number(num));
  return check !== -1;
};

export const formatDate = timeFormat('%b %d, \'%y');

export const getDate = (d) => new Date(d.date);

export const getStockValue = (d) => d.close;

export const bisectDate = bisector(d => new Date(d.date)).left;

export const getWindowDimensions = () => {
  const {innerWidth: width, innerHeight: height} = window;
  return {
    width,
    height
  };
};

export const dateFormat = (date, format = 'DD MMM, YYYY') => {
  return moment(date).format(format);
};

export const findTestByAttr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const checkProps = (component, expectedProps) => {
  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

