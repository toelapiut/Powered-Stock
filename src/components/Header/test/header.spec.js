import React from 'react';
import Header from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {findTestByAttr, checkProps} from '../../../helper/utils';


const setUp = (props = {}) => {
  return shallow(<Header {...props}/>);
};

let props = {
  date: {
    start: '01-04-2020',
    end: '01-04-2020'
  },
  name: 'Powered By People Inc',
  ticker: 'PBP',
  start: '01-04-2020',
  end: '01-04-2020',
  isOpen: false,
  onChangeDates: () => null,
  onOpenCalendar: () => null,
};


describe('Header Component', () => {
  describe('Component renders tests', () => {
    let component;
    beforeAll(() => {
      component = setUp(props);
    });

    test('should render snapshot correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render without error', () => {
      const wrapper = findTestByAttr(component, 'header-container');
      expect(wrapper.length).toBe(1);
    });
  });

  describe('Check property types', () => {
    it('should not throw a warning', () => {
      const propErr = checkProps(Header, props);
      expect(propErr).toBeUndefined();
    });
  });
});