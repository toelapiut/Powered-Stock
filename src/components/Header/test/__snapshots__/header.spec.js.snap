// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Component renders tests should render snapshot correctly 1`] = `
<div
  className="container"
  data-test="header-container"
>
  <div>
    <h1
      className="ticker"
    >
      PBP
    </h1>
    <p
      className="name"
    >
      Powered By People Inc
    </p>
  </div>
  <DateInput
    date={
      Object {
        "end": "01-04-2020",
        "start": "01-04-2020",
      }
    }
    end="01-04-2020"
    isOpen={false}
    onChangeDates={[Function]}
    onOpenCalendar={[Function]}
    start="01-04-2020"
  />
</div>
`;
